junit-platform-surefire-provider/src/test/java/org/junit/platform/surefire/provider/RunListenerAdapterTests.java

```diff
@@ -92,13 +95,17 @@ void notifiedWhenMethodExecutionSkipped() throws Exception {
 	@Test
 	void notifiedWithCorrectNamesWhenClassExecutionSkipped() throws Exception {
 		ArgumentCaptor<ReportEntry> entryCaptor = ArgumentCaptor.forClass(ReportEntry.class);
+		TestPlan testPlan = TestPlan.from(Collections.singletonList(new EngineDescriptor(newId(), "Luke's Plan")));
+		adapter.testPlanExecutionStarted(testPlan);
+
+		TestIdentifier classIdentifier = newClassIdentifierOnTestPlan(testPlan);
 
-		adapter.executionSkipped(newClassIdentifier(), "test");
+		adapter.executionSkipped(classIdentifier, "test");
 		verify(listener).testSkipped(entryCaptor.capture());
 
 		ReportEntry entry = entryCaptor.getValue();
 		assertTrue(MyTestClass.class.getTypeName().contains(entry.getName()));
-		assertEquals(MyTestClass.class.getName(), entry.getSourceName());
+		assertEquals("<unrooted>", entry.getSourceName());
```

Comment:
Checking the old behaviour of surefire reporting an entire class being skipped it gives: 
<testcase name="org.test.SureFireTest" classname="org.test.SureFireTest" time="0">
    <skipped/>
</testcase>

And running this pr through an integration test we report:
name= "org.test.SureFireTest" and classname= "JUnit Jupiter"